tools/benchmark/src/mocha/memoryTestRunner.ts

```diff
@@ -296,6 +310,17 @@ export function benchmarkMemory(testObject: IMemoryTestObject): Test {
 						heapUsedStats.marginOfErrorPercent > args.maxRelativeMarginOfError
 					);
 
+					const avgHeapUsed = heapUsedStats.arithmeticMean;
+					const allowedMemoryUsage =
+						args.baselineMemoryUsage * (1 + args.allowedDeviation / 100);
+					const tolerance = args.baselineMemoryUsage * (args.allowedDeviation / 100);
+
+					if (ENABLE_MEM_REGRESSION && avgHeapUsed > allowedMemoryUsage) {
```

Comment:
I think it'd be nice if we checked separately for variation over or under. If it goes over, it's probably a regression. If it goes under, it might be a performance improvement that should be accompanied by an update to the baseline? And the error message should make that clear.